src/tap/cmd.rs

```diff
@@ -98,12 +99,27 @@ async fn run(
         );
     };
 
+    let start_time = Instant::now();
+    let stream_duration = match opts.duration {
+        Some(duration) => Duration::from_millis(duration),
+        None => Duration::MAX,
+    };
+
     // Loop over the returned results, printing out tap events.
     #[allow(clippy::print_stdout)]
     #[allow(clippy::print_stderr)]
     loop {
-        let message = stream.next().await;
-        if let Some(Some(res)) = message {
+        if start_time.elapsed() >= stream_duration {
+            return exitcode::OK;
+        }
+
+        let message = 
+            timeout(stream_duration - start_time.elapsed(), stream.next()).await;
+        if message.is_err() {
+            return exitcode::OK;
```

Comment:
Is this right, that we just return immediately with no warning if the message is in error?

Reply:
Yea, idea was that this will timeout if it reaches the duration specified by the user, to handle a case where the current interval lasts longer than the duration (an extreme example would be that the duration is 5 seconds but the interval is 1 minute), in which case the time `tap` would be running for 1 minute before exiting even though technically the duration is less.

The other option would be to just leave only the `if start_time.elapsed() >= stream_duration` check, it depends on whether we think the case I described above is something we want to cover.